back/engines/commercial/analytics/spec/services/analytics/query_runner_service_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'query'
+
+describe Analytics::QueryRunnerService do
+  describe 'run' do
+    it 'return the ID field for each post' do
+      ideas = create_list(:idea, 5)
+      create(:dimension_type_idea)
+      query_param = ActionController::Parameters.new(fact: 'post', fields: 'id')
```

Comment:
We could just use a hash here.

Reply:
we are using `.to_unsafe_hash` in the validation thats why it needs to be a Params object